ethereumj-core/src/main/java/org/ethereum/sharding/validator/BeaconProposerImpl.java

```diff
@@ -74,10 +81,25 @@ public BeaconProposerImpl(Randao randao, StateRepository repository,
     @Override
     public Beacon createNewBlock(Input in, byte[] pubKey) {
         Beacon block = new Beacon(in.parent.getHash(), randaoReveal(in.state, pubKey), in.mainChainRef,
-                HashUtil.EMPTY_DATA_HASH, in.slotNumber);
+                HashUtil.EMPTY_DATA_HASH, in.slotNumber, new AttestationRecord[0]);
         BeaconState newState = stateTransition.applyBlock(block, in.state);
         block.setStateHash(newState.getHash());
 
+        AttestationRecord[] attestationRecords = new AttestationRecord[1];
```

Comment:
> Verify that an attestation from this validator is part of the first (ie. item 0 in the array) AttestationRecord object; this attester can be considered to be the proposer of the parent block. In general, when a block is produced, it is broadcasted at the network layer along with the attestation from its proposer.

I was totally wrong that proposer should _include_ this attestation to the block it proposes. As to the spec it should broadcast block and zero attestation alongside to the network. And its ancestor should include this attestation at the first position in the array of attestations. This totally makes sense, cause including zero attestation into the same block will inevitably change its hash and then the attestation won't pass verification part.

Reply:
Also, zero attestation should be a part of `ValidatorService` rather than placed here